src/main/kotlin/link/infra/packwiz/installer/UpdateManager.kt

```diff
@@ -97,6 +99,105 @@ class UpdateManager internal constructor(private val opts: Options, val ui: IUse
+			}
+
+			var manifestModified = false
+			val modLoaders = hashMapOf("net.minecraftforge" to "forge", "net.fabricmc.fabric-loader" to "fabric")
+			val modLoadersClasses = modLoaders.entries.associate{(k,v)-> v to k}
+			var modLoaderFound = false
+			var modLoader: String? = null
+			var modLoaderOldVer: String? = null
+			var mcOldVer = "Unknown"
+			val components = multimcManifest["components"].asJsonArray
+			for (componentObj in components) {
+				val component = componentObj.asJsonObject
+
+				val version = component["version"].asString
+				when (component["uid"].asString) {
+					"net.minecraft" -> {
+						mcOldVer = version
+						if (version != pf.versions?.get("minecraft")) {
+							manifestModified = true
+							component.addProperty("version", pf.versions?.get("minecraft"))
+						}
+					}
+				}
+				// If we find any of the modloaders we support, we save it and check the version
+				if (modLoaders.containsKey(component["uid"].asString)) {
+					modLoaderFound = true
+					modLoader = modLoaders.getValue(component["uid"].asString)
+					modLoaderOldVer = version
+					if (version != pf.versions?.get(modLoader)) {
+						manifestModified = true
+						component.addProperty("version", pf.versions?.get(modLoader))
+					}
+				}
+			}
+
+			// If we can't find the mod loader in the MultiMC file, we add it
+			if (!modLoaderFound) {
+				for ((loaderClass, loader) in modLoaders) {
+					if (pf.versions?.containsKey(loader) ?: false) {
+						modLoader = loader
+						break
+					}
+				}
+				// If we can't find it in the modpack pack file, something is wrong, we'll stop here
+				if (modLoader == null) ui.showErrorAndExit("Mod Loader not found in modpack pack file")
+
+				components.add(gson.toJsonTree(hashMapOf("uid" to modLoadersClasses.get(modLoader), "version" to pf.versions?.get(modLoader))))
+				manifestModified = true
+			}
+
+			if (manifestModified) {
+				// The manifest has been modified, so before saving it we'll ask the user
+				// if they wanna update it, continue without updating it, or exit
+				val oldVers = listOf(Pair("minecraft", mcOldVer), Pair(modLoader!!, modLoaderOldVer))
+				val newVers = listOf(Pair("minecraft", pf.versions?.getValue("minecraft")), Pair(modLoader!!, pf.versions?.getValue(modLoader!!)))
+
+				when (ui.showUpdateConfirmationDialog(oldVers, newVers)) {
+					IUserInterface.UpdateConfirmationResult.CANCELLED -> {
+						cancelled = true
+					}
+					IUserInterface.UpdateConfirmationResult.CONTINUE -> {
+						cancelledStartGame = true
+					}
+				}
+				handleCancellation()
+
+				multimcManifestFile.writeText(gson.toJson(multimcManifest))
+				Log.info("Updated modpack Minecraft and/or ${modLoader?.replaceFirstChar { it.uppercase() }} version")
+			}
+
+			if (ui.cancelButtonPressed) {
+				showCancellationDialog()
+				handleCancellation()
+			}
+		} else {
+			Log.warn("MultiMC installation not detected... Tried looking for it in $multimcManifestPath")
```

Comment:
Shouldn't be a warning, packwiz-installer does not need to be run under MultiMC (can be run standalone, I'm working on vanilla launcher support)